src/GiGraph.Dot/Extensions/DotGraphExtension.cs

```diff
@@ -32,21 +33,17 @@ public static class DotGraphExtension
         ///     The syntax rules to use.
         /// </param>
         /// <param name="encoding">
-        ///     The encoding to use for the output text.
+        ///     Unused. Kept for backward compatibility.
         /// </param>
         public static string Build(this DotGraph graph, DotFormattingOptions formattingOptions = null,
             DotSyntaxOptions syntaxOptions = null, DotSyntaxRules syntaxRules = null, Encoding encoding = null)
         {
-            var output = new MemoryStream();
-
-            using var outputWriter = encoding is not null
-                ? new StreamWriter(output, encoding)
-                : new StreamWriter(output);
+            var initialCapacity = (graph.Nodes.Count + graph.Edges.Count) * 16;
```

Comment:
It is actually more complex than that, and I'd probably not specify the initial capacity as it is hard to estimate it. The graph would have to be traversed recursively to know how many nodes and edges there actually are. The thing is that the graph may be composed of subgraphs and clusters, but also there are 'subsections', being parts of the output, each of which may as well contain edges, nodes, subgraphs and clusters.

Reply:
I'm aware that's it's a very very rough estimate. I thought giving an initial capacity would be better than 0 which will automatically grow at the first write. But I must admit I haven't benchmarked anything and that not specifying any initial capacity might be totally fine.